### ISSUE TITLE
[Task]: Consider supporting old clients for hard->soft blocks

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons#15208]
### Description

This hard->soft transition is out of scope but still something AMO technically supports. Per https://docs.google.com/document/d/1JmGUPD9fZ8ri_xdF1zBwOXsQHfBX-51fV2HYbBDsFQg/edit?usp=sharing, we could use a "super-mega-unified stash" to support any clients:

```json
{
  "blocked": [], 
  "softblocked": ["@addon:version"],
  "unblocked": ["@addon:version"]
}
```

Old clients not support soft-blocking would see the version as unblocked (because they ignore properties they don't recognize, and they don't recognize `softblocked` in the stash), and new clients (that support soft-blocking) will see the version as soft-blocked.

This works fine, except that the current AMO implementation generates both filters and stashes at the exact same time. In a previous iteration for https://github.com/mozilla/addons/issues/15014, it was possible to have a new soft filter generated, and a stash - leaving the version in the `unblocked` list of the stash only. That is a problem because new clients will see the version as "unblocked" and - because the data in the filter isn't newer than the stash (same timestamp) - these clients will consider the version as unblocked instead of soft-blocked.

We need to think about how we want to improve the situation for old clients, if possible. Otherwise, we'll need to document that old clients will never unblock a hard block turned into a soft block.

### Acceptance Criteria

```[tasklist]
  ### Milestones/checkpoints
  - [ ] support old clients
```

### Checks

- [X] If I have identified that the work is specific to a repository, I have removed "repository:addons-server" or "repository:addons-frontend"

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/AMOENG-1272)
